openapi_python_client/templates/property_templates/union_property.py.jinja

```diff
@@ -1,3 +1,36 @@
+{% macro construct_inner_property(inner_property) %}
+{% import "property_templates/" + inner_property.template as inner_template %}
+{% if inner_template.check_type_for_construct %}
+if not {{ inner_template.check_type_for_construct(inner_property, "data") }}:
+    raise TypeError()
+{% endif %}
+{{ inner_template.construct(inner_property, "data") }}
+return {{ inner_property.python_name }}
+{%- endmacro %}
+
+{% macro construct_discriminator_lookup(property) %}
+{% set _discriminator_properties = [] -%}
+{% for discriminator in property.discriminators %}
+{{- _discriminator_properties.append(discriminator.property_name) or "" -}}
+if not isinstance(data, dict):
```

Comment:
As described in comments in `union.py`, there can be multiple discriminators if there was a nested union. The logic here tries each discriminator property in order; if that property exists, it checks the value, and if it finds a matching class, it uses that class. If no matches were found for any discriminator then it raises an error.